### PR TITLE
[SNS] - adds CreatePlatformEndpoint and DeleteEndpoint

### PR DESCRIPTION
**Pull request by smoench**
We have a use case for un-/register devices/mobile apps on platform applications.

### CODE REVIEW

File: src/Service/Sns/tests/Unit/Input/CreatePlatformEndpointInputTest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        self::fail('Not implemented');
+
+        $input = new CreatePlatformEndpointInput([
+            'PlatformApplicationArn' => 'change me',
+            'Token' => 'change me',
+            'CustomUserData' => 'change me',
+            'Attributes' => ['change me' => 'change me'],
+        ]);
+
+        // see https://docs.aws.amazon.com/sns/latest/APIReference/API_CreatePlatformEndpoint.html
```
Comment (jderusse) [inline src/Service/Sns/tests/Unit/Input/CreatePlatformEndpointInputTest.php:21]: right link is https://docs.aws.amazon.com/sns/latest/api/API_CreatePlatformEndpoint.html

Comment (jderusse) [reply src/Service/Sns/tests/Unit/Input/CreatePlatformEndpointInputTest.php:21]: and you can use the snippet provided in the doc:
```
https://sns.us-west-2.amazonaws.com/?Action=CreatePlatformEndpoint
&PlatformApplicationArn=arn%3Aaws%3Asns%3Aus-west-2%3A123456789012%3Aapp%2FGCM%2Fgcmpushapp
&Token=APA91bGi7fFachkC1xjlqT66VYEucGHochmf1VQAr9k...jsM0PKPxKhddCzx6paEsyay9Zn3D4wNUJb8m6HZrBEXAMPLE
&CustomUserData=UserId%3D27576823
&Version=2010-03-31
&AUTHPARAMS 
```

Comment (Nyholm) [reply src/Service/Sns/tests/Unit/Input/CreatePlatformEndpointInputTest.php]: Maybe it could. But I suggest doing that in a second PR.

**Comment by smoench** [comment]
Tests are green :-) Thank you Jérémy and Tobias for this awesome and neat client implementation

File: src/Service/Sns/tests/Unit/Input/CreatePlatformEndpointInputTest.php (hunk 1 above)
Comment (jderusse) [reply src/Service/Sns/tests/Unit/Input/CreatePlatformEndpointInputTest.php]: Sadly AWS is inconsistent in naming. And, as far as I know, the link to the Documentation cannot be guessed.
hegenerated link works for several services, but not all. :(

Comment (smoench) [reply src/Service/Sns/tests/Unit/Input/CreatePlatformEndpointInputTest.php]: Could they be added to the manifest.json?

Comment (jderusse) [reply src/Service/Sns/tests/Unit/Input/CreatePlatformEndpointInputTest.php]: yes I think that would be the best solution.

Comment (smoench) [reply src/Service/Sns/tests/Unit/Input/CreatePlatformEndpointInputTest.php]: I added an issue #687

**Pull request merged**